File: problems/cpp/max-increase-to-keep-city-skyline.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int n = grid.size();
        vector<vector<int>>v = grid;
        vector<int>mc(n, 0), mr(n, 0);
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                mc[j] = max(mc[j], grid[i][j]);
                mr[i] = max(mr[i], grid[i][j]);
            }
        }
        int ans = 0;
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                ans += min(mc[j], mr[i]) - grid[i][j];
            }
        }
        return ans;
    }
};

File: problems/python/max-increase-to-keep-city-skyline.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        n = len(grid)
        m = len(grid[0])
        mc = [0 for i in range(m)]
        mr = [0 for i in range(n)]
        ans = 0
        for i in range(n):
            for j in range(m):
                mc[j] = max(mc[j], grid[i][j])
                mr[i] = max(mr[i], grid[i][j])
        for i in range(n):
            for j in range(m):
                ans += min(mr[i], mc[j]) - grid[i][j]
        return ans